src/canmatrix/canmatrix.py

```diff
@@ -1183,50 +1212,50 @@ class CanMatrix(object):
     valueTables (global defined values)
     """
 
-    attributes = attr.ib(type=dict, factory=dict)
-    ecus = attr.ib(type=list, factory=list)
-    frames = attr.ib(type=list, factory=list)
+    attributes = attr.ib(factory=dict)  # type: typing.MutableMapping[str, typing.Any]
+    ecus = attr.ib(factory=list)  # type: typing.MutableSequence[Ecu]
+    frames = attr.ib(factory=list)  # type: typing.MutableSequence[Frame]
 
-    signal_defines = attr.ib(type=dict, factory=dict)
-    frame_defines = attr.ib(type=dict, factory=dict)
-    global_defines = attr.ib(type=dict, factory=dict)
-    env_defines = attr.ib(type=dict, factory=dict)
-    ecu_defines = attr.ib(type=dict, factory=dict)
-    value_tables = attr.ib(type=dict, factory=dict)
-    env_vars = attr.ib(type=dict, factory=dict)
-    signals = attr.ib(type=list, factory=list)
+    signal_defines = attr.ib(factory=dict)  # type: typing.MutableMapping[str, Define]
+    frame_defines = attr.ib(factory=dict)  # type: typing.MutableMapping[str, Define]
+    global_defines = attr.ib(factory=dict)  # type: typing.MutableMapping[str, Define]
+    env_defines = attr.ib(factory=dict)  # type: typing.MutableMapping[str, Define]
+    ecu_defines = attr.ib(factory=dict)  # type: typing.MutableMapping[str, Define]
+    value_tables = attr.ib(factory=dict)  # type: typing.MutableMapping[str, typing.MutableMapping]
+    env_vars = attr.ib(factory=dict)  # type: typing.MutableMapping[str, typing.MutableMapping]
+    signals = attr.ib(factory=list)  # type: typing.MutableSequence[Signal]
 
-    load_errors = attr.ib(type=list, factory=list)
+    load_errors = attr.ib(factory=list)  # type: typing.MutableSequence[Exception]
 
-    def __iter__(self):
+    def __iter__(self):  # type: () -> typing.Iterator
```

Comment:
```suggestion
    def __iter__(self):  # type: () -> typing.Iterator[Frame]
```

Reply:
Thank you, there will be more such details I guess. Once having the file fully annotated I'll check it once again applying the experiences I gathered in the meantime.